File: components/vehicle-domain/src/main/java/com/vmware/tanzu/data/IoT/vehicles/domains/VehicleTelemetry.java
package com.vmware.tanzu.data.IoT.vehicles.domains;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@EqualsAndHashCode(callSuper = true)
@Data
@AllArgsConstructor
@NoArgsConstructor
public class VehicleTelemetry extends Vehicle {

    private LocalDateTime captureTimestamp  = LocalDateTime.now();

    public VehicleTelemetry(Vehicle vehicle) {
        this.setId(vehicle.getId());
        this.setGpsLocation(vehicle.getGpsLocation());
        this.setVin(vehicle.getVin());
        this.setSpeed(vehicle.getSpeed());
        this.setOdometer(vehicle.getOdometer());
        this.setTemperature(vehicle.getTemperature());
    }
}

File: components/IoT-connected-vehicles-domains/src/main/kotlin/com/vmware/tanzu/data/IoT/vehicles/domains/VehicleTelemetry.kt
package com.vmware.tanzu.data.IoT.vehicles.domains

import java.time.LocalDateTime

/**
 * @author Gregory Green
 */
data class VehicleTelemetry  (
    override var vin: String = "",
    override var odometer: Long = 0,
    override var speed : Int = 0,
    override var temperature: Int = 0,
    override var gpsLocation: GpsLocation? = null,
    var captureTimestamp : LocalDateTime = LocalDateTime.now(),
    var id : Long = 0
) : IVehicle